File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        int left = 0;
        int right = numbers.size() -1;
        int currentSum = 0;
        
        while(left < right)
        {
            currentSum = numbers[left] + numbers[right];

            if(currentSum == target)
            {
                return {left + 1, right + 1};
            }
            else if(currentSum > target)
            {
                right--;
            }
            else
            {
                left++;
            }
        }

        return {};
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        ROWS, COLS = len(matrix), len(matrix[0]) # get the number of ROWS and the numbers of elements in the first row (aka the number of COLS)

        l, r = 0, ROWS* COLS - 1 # start of the flattened matrix, end of the flattened matrix
        while l <= r:
            m = l + (r-l) // 2 # get the midpoint
            row, col = m // COLS, m % COLS # get the 2D coordinates 

            if target < matrix[row][col]:
                r = m -1
            elif target > matrix[row][col]:
                l = m + 1
            else:
                return True
        return False